pkg/provisioner/ironic/ironic.go

```diff
@@ -1176,9 +1176,15 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 func (p *ironicProvisioner) changePower(ironicNode *nodes.Node, target nodes.TargetPowerState) (result provisioner.Result, err error) {
 	p.log.Info("changing power state")
 
-	// If we're here, we're going to change the state and we should
-	// wait to try that again.
-	result.RequeueAfter = powerRequeueDelay
+	if ironicNode.TargetProvisionState != "" {
```

Comment:
Are you sure it's an empty string? Ironic API returns JSON `null` in this case.

Reply:
The go type defined by the client is a string, not a pointer to a string. It can't be nil.